### PR TITLE
CTP-2529 - Show recent courses even if hidden or not enrolled

### PR DESCRIPTION
**Pull request by stuartlamour**
Swap back to db query as core function doesn't do what we need it to.

### CODE REVIEW

File: block_pickup.php [hunk 1]
```diff
@@ -117,10 +117,22 @@ public function fetch_recent_mods() : array {
...
+        global $USER, $DB;
 
         // Get recent courses.
-        $courserecords = course_get_recent_courses($USER->id, 3);
+         $sql = "SELECT c.id, c.fullname, c.visible, cc.name as catname
+                  FROM {user_lastaccess} ula
+                  JOIN {course} c ON c.id = ula.courseid
+                  JOIN {course_categories} cc ON cc.id = c.category
+                 WHERE ula.userid = :userid
+              ORDER BY ula.timeaccess DESC
```
Comment (aydevworks) [inline block_pickup.php]: Looks like the lines are not aligned correctly.

**Review by aydevworks — COMMENTED** [review]
Looks fine to me, except the lines alignment. Will let @aspark21 to do the final approval.

File: block_pickup.php [hunk 2]
```diff
@@ -134,7 +146,8 @@ public function fetch_recent_courses() : array {
             $course = new stdClass();
             $course->fullname = $cr->fullname;
             $course->viewurl = new moodle_url('/course/view.php', array('id' => $cr->id));
-            $course->coursecategory = core_course_category::get($cr->category)->get_formatted_name();
+            $course->visible = $cr->visible;
+            $course->coursecategory = $cr->catname;
```
Comment (aspark21) [inline block_pickup.php:150]: why are we looking up the course category at all?

Comment (aspark21) [reply block_pickup.php:150]: ah, I can see it's listed under the course name, hadn't noticed before

File: block_pickup.php [hunk 3]
```diff
@@ -117,10 +117,22 @@ public function fetch_recent_mods() : array {
...
+                   JOIN {course_categories} cc ON cc.id = c.category
+                  WHERE ula.userid = :userid
+               ORDER BY ula.timeaccess DESC
+                  LIMIT 3";
+
+        $params = array(
+            'userid' => $USER->id,
+        );
+
+        $courserecords = $DB->get_records_sql($sql, $params);
```
Comment (aspark21) [inline block_pickup.php:135]: Can we check on the error handling of this vs the core function please

@aydevworks @stuartlamour 

Fine if we want to display courses which are hidden from students to all but can we avoid the issues we had with activity types that weren't error handled

File: block_pickup.php (hunk 2 above)
Comment (aspark21) [reply block_pickup.php:150]: course category could be fixed as Alex suggested by changing to this
$course->coursecategory = core_course_category::get($cr->category, 'IGNORE_MISSING', true);

but I get we're now getting catname through the db query above instead

**Review by aspark21 — APPROVED** [review]

**Pull request merged**